### ISSUE TITLE
How to avoid replacing some `?` in bindValue 

### ISSUE THREAD

**Issue by zion03** [doctrine/dbal#3078]
I have query fragment `WHERE n.attributes ? ?`
and then execute 
`$statement = $this->getEntityManager()->getConnection()->prepare($sql);`
with params 
```
foreach($sqlArr['params'] as $id => $val) {
 $statement->bindValue($id+1, $val);
}
```

Is there way to quote first `?` as it's a PostgreSQL JSONB operator (not pleaceholder). I assume the fasted way will go to  named prepared statements but it's not easy in my source.

**Comment by Ocramius** [comment]
Parameter binding is only for values. Operators are not values, so they (usually) cannot be bound to parameters, but that depends to what the PostgreSQL client library can do.

You'll probably need either

https://github.com/doctrine/dbal/blob/f76bf5ef631cec551a86c2291fc749534febebf1/lib/Doctrine/DBAL/Connection.php#L842-L857

or

https://github.com/doctrine/dbal/blob/f76bf5ef631cec551a86c2291fc749534febebf1/lib/Doctrine/DBAL/Connection.php#L823-L840

**Comment by Ocramius** [comment]
> DBAL must ignore `?` in some keys.

https://stackoverflow.com/questions/30629076/how-to-escape-the-question-mark-operator-to-query-postgresql-jsonb-type-in-r?rq=1

**Comment by zion03** [comment]
Okay, I understand.
Is there a way to execute raw query without replacing values or any other checking on `PDO` using` DBAL` connection as is.
I have a big query and much better for me prepare it by myself.
i.e:

> SELECT .. FROM .. WHERE n.categories_id @> 1 AND n.attributes::jsonb ? '123' AND ...

And executing it as is

**Comment by zion03** [comment]
Sorry, look like problem in a `PDO` driver

**Comment by zion03** [comment]
Solution: https://stackoverflow.com/questions/30461558/jsonb-existential-operators-with-parameterised-queries

**Comment by drola** [comment]
This will be fixed in PHP 7.4: https://wiki.php.net/rfc/pdo_escape_placeholders
